### PR TITLE
Update Reolink supported model list

### PR DESCRIPTION
**Pull request by starkillerOG**
## Proposed change

Show which models are discontinued and not sold anymore, those models will continue to work with HA.
Add links for each of the models to their product pages on the reolink.com site.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/reolink.markdown [hunk 1]
```diff
@@ -166,37 +166,43 @@ The latest firmware can be downloaded from the [Reolink download center](https:/
...
-- RLN8-410 NVR
-- RLN16-410 NVR
-- RLN36 NVR
-- Reolink Duo 2 WiFi
-- Reolink Duo Floodlight PoE
-- Reolink TrackMix (PoE and Wi-Fi)
-- Reolink Video Doorbell (PoE and Wi-Fi)
+- C1 Pro*
+- C2 Pro*
+- [CX410](https://reolink.pxf.io/bamgZv)
```
Comment (frenck) [inline source/_integrations/reolink.markdown]: These are affiliate links, please remove those.

**Review by frenck — DISMISSED** [review]

File: source/_integrations/reolink.markdown (hunk 1 above)
Comment (starkillerOG) [reply source/_integrations/reolink.markdown]: That is a pitty that it is not allowed...
Have changed them to normal links
and removed the disclaimer explaining it were affiliate links

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @starkillerOG! :+1:

**Review by klaasnicolaas — APPROVED** [review]
✅ LGTM

**Pull request merged**